pkg/controller/prpqr_reconciler/prpqr_reconciler.go

```diff
@@ -225,12 +258,70 @@ func (r *reconciler) reconcile(ctx context.Context, req reconcile.Request, logge
 	return nil
 }
 
-func constructCondition(createdJobs map[string]v1.PullRequestPayloadJobStatus) metav1.Condition {
+func reconcileStatus(theirs *v1.PullRequestPayloadQualificationRun, ourStatuses map[string]*v1.PullRequestPayloadJobStatus, ourCondition metav1.Condition) {
+	var foundCondition bool
+	for i := range theirs.Status.Conditions {
+		if theirs.Status.Conditions[i].Type == ourCondition.Type {
+			if theirs.Status.Conditions[i].Status != ourCondition.Status {
+				theirs.Status.Conditions[i] = ourCondition
+			}
+			foundCondition = true
+			break
+		}
+	}
+	if !foundCondition {
+		theirs.Status.Conditions = append(theirs.Status.Conditions, ourCondition)
+	}
+
+	statusByJobName := map[string]*v1.PullRequestPayloadJobStatus{}
+	for i := range theirs.Status.Jobs {
+		jobName := theirs.Status.Jobs[i].ReleaseJobName
+		statusByJobName[jobName] = &theirs.Status.Jobs[i]
+	}
+
+	theirs.Status.Jobs = []v1.PullRequestPayloadJobStatus{}
+	for _, spec := range theirs.Spec.Jobs.Jobs {
+		jobName := spec.JobName(jobconfig.PeriodicPrefix)
+		our := ourStatuses[jobName]
+		their := statusByJobName[jobName]
+		theirs.Status.Jobs = append(theirs.Status.Jobs, reconcileJobStatus(jobName, their, our))
+	}
+}
+
+func reconcileJobStatus(name string, their, our *v1.PullRequestPayloadJobStatus) v1.PullRequestPayloadJobStatus {
```

Comment:
I don't think that we need to reconcile the job statuses, since they will always be empty when we will reach this code.